proxy/context.go

```diff
@@ -44,84 +44,68 @@ type Environment struct {
 	Mesh *proxyconfig.ProxyMeshConfig
 }
 
-// Role declares the proxy node role in the mesh
-type Role interface {
-	// nolint: megacheck
-	isProxyRole()
+// Role defines the proxy role
+type Role struct {
```

Comment:
This is overloading the term with more info. Suggest using a different noun like router or ProxyAttributes or something